### PR TITLE
Add `--resource-type` flag to `policy_sentry query action-table` command

### PR DESCRIPTION
**Pull request by reetasingh**
## What does this PR do?

**Address https://github.com/salesforce/policy_sentry/issues/255 with the Bonus Points**

[//]: # (Required: Describe the effects of your pull request in detail. If
multiple changes are involved, a bulleted list is often useful.)

1. Adding **--resource-type** flag to `policy_sentry query action-table`
2. Removing **--wildcard-only** flag from `policy_sentry query action-table`
3. Adding method `get_actions_matching_arn_type`
4. Modifying method `get_actions_with_arn_type_and_access_level` 
4. Adding unit tests for the new methods added and methods modified
5. Adding integration tests for new -`-resource-type` flag

## What gif best describes this PR or how it makes you feel?

[//]: # (Encouraged: Insert an appropriate gif/meme to brighten up your reviewer's day.)
ALT

## Completion checklist

- [x] CHANGELOG.md has been updated to reflect the changes
- [x] Additions and changes have unit tests
- [x] [Unit tests, Pylint, security testing, and Integration tests are passing.](https://github.com/salesforce/policy_sentry/actions). GitHub actions does this automatically
- [x] The pull request has been appropriately labeled using the provided PR labels

### CODE REVIEW

File: policy_sentry/querying/actions.py [hunk 1]
```diff
@@ -161,50 +157,103 @@ def get_actions_at_access_level_that_support_wildcard_arns_only(
...
+                for resource_type in action_data["resource_types"]:
+                    if resource_type == '':
+                        results.append(f"{service_prefix}:{action_name}")
     return results
 
 
-def get_actions_with_arn_type_and_access_level(
-    service_prefix, resource_type_name, access_level
-):
+def get_actions_matching_arn_type(service_prefix, resource_type_name):
```
Comment (reetasingh) [inline policy_sentry/querying/actions.py:230]: new method added

**Comment by reetasingh** [comment]
@kmcquade let me know your thoughts on this PR

File: policy_sentry/querying/actions.py [hunk 2]
```diff
@@ -161,50 +157,103 @@ def get_actions_at_access_level_that_support_wildcard_arns_only(
...
+        resource_type_name: The ARN type name, like `bucket` or `key`
+        access_level: Access level like "Read" or "List" or "Permissions management"
+    Return:
+        List: A list of actions that have that ARN type and Access level
+    """
+    service_prefix_data = get_service_prefix_data(service_prefix)
+    results = []
+
+    if resource_type_name == '*':
+        return get_actions_at_access_level_that_support_wildcard_arns_only(service_prefix, access_level)
```
Comment (reetasingh) [inline policy_sentry/querying/actions.py:177]: for resource_type_name == '*' calling the existing get_actions_at_access_level_that_support_wildcard_arns_only method

File: policy_sentry/querying/actions.py [hunk 3]
```diff
@@ -161,50 +157,103 @@ def get_actions_at_access_level_that_support_wildcard_arns_only(
...
     Arguments:
         service_prefix: A single AWS service prefix, like `s3` or `kms`
         resource_type_name: The ARN type name, like `bucket` or `key`
-        access_level: Access level like "Read" or "List" or "Permissions management"
     Return:
-        List: A list of actions that have that ARN type and Access level
+        List: A list of actions that have that ARN type
     """
+    if resource_type_name == '*':
+        return get_actions_that_support_wildcard_arns_only(service_prefix)
```
Comment (reetasingh) [inline policy_sentry/querying/actions.py:241]: for resource_type_name == '*' calling the existing get_actions_that_support_wildcard_arns_only method

Comment (reetasingh) [reply policy_sentry/querying/actions.py:241]: in this way the get_actions_matching_arn_type will support both the case where resource_type_name='*' or not '*'

File: policy_sentry/querying/actions.py [hunk 4]
```diff
@@ -161,50 +157,103 @@ def get_actions_at_access_level_that_support_wildcard_arns_only(
...
+
     service_prefix_data = get_service_prefix_data(service_prefix)
     results = []
 
-    for action_name, action_data in service_prefix_data["privileges"].items():
-        if action_data["access_level"] == access_level:
+    if service_prefix == "all":
```
Comment (reetasingh) [inline policy_sentry/querying/actions.py:246]: if resource_type_name !='*'

File: policy_sentry/querying/actions.py [hunk 5]
```diff
@@ -161,50 +157,103 @@ def get_actions_at_access_level_that_support_wildcard_arns_only(
...
+        access_level: Access level like "Read" or "List" or "Permissions management"
+    Return:
+        List: A list of actions that have that ARN type and Access level
+    """
+    service_prefix_data = get_service_prefix_data(service_prefix)
+    results = []
+
+    if resource_type_name == '*':
+        return get_actions_at_access_level_that_support_wildcard_arns_only(service_prefix, access_level)
+
```
Comment (reetasingh) [inline policy_sentry/querying/actions.py:178]: if resource_type_name !='*' this block will be executed

**Comment by reetasingh** [comment]
OutPut:
```
(venv) reetasingh-ltm8:policy_sentry reetasingh$ policy_sentry query action-table  --help
Usage: policy_sentry query action-table [OPTIONS]

  Query the Action Table from the Policy Sentry database

Options:
  --service TEXT                  Filter according to AWS service.  [required]
  --name TEXT                     The name of IAM Action. For example, if the
                                  action is "iam:ListUsers", supply
                                  "ListUsers" here.

  --access-level [read|write|list|tagging|permissions-management]
                                  Filter according to CRUD levels. Acceptable
                                  values are read, write, list, tagging,
                                  permissions-management

  --condition TEXT                Supply a condition key to show a list of all
                                  IAM actions that support the condition key.

  --resource-type TEXT            Supply a resource type to show a list of all
                                  IAM actions that support the resource type.

  --fmt [yaml|json]               Format output as YAML or JSON. Defaults to
                                  "yaml"

  -v, --verbosity LVL             Either CRITICAL, ERROR, WARNING, INFO or
                                  DEBUG

  --help                          Show this message and exit.
(venv) reetasingh-ltm8:policy_sentry reetasingh$ 
```

**Comment by reetasingh** [comment]
```
(venv) reetasingh-ltm8:policy_sentry reetasingh$ policy_sentry query action-table --service ssm --access-level write --resource-type parameter
Using the Local IAM definition: /Users/reetasingh/.policy_sentry/iam-definition.json. To leverage the bundled definition instead, remove the folder $HOME/.policy_sentry/
ssm WRITE actions that have the resource type PARAMETER:
[
    "ssm:DeleteParameter",
    "ssm:DeleteParameters",
    "ssm:LabelParameterVersion",
    "ssm:PutParameter"
]
(venv) reetasingh-ltm8:policy_sentry reetasingh$ policy_sentry query action-table --service secretsmanager --access-level list --resource-type '*'
Using the Local IAM definition: /Users/reetasingh/.policy_sentry/iam-definition.json. To leverage the bundled definition instead, remove the folder $HOME/.policy_sentry/
secretsmanager LIST actions that have the resource type *:
[
    "secretsmanager:ListSecrets"
]
```

**Comment by reetasingh** [comment]
```
(venv) reetasingh-ltm8:policy_sentry reetasingh$ policy_sentry query action-table --service secretsmanager  --resource-type '*'
Using the Local IAM definition: /Users/reetasingh/.policy_sentry/iam-definition.json. To leverage the bundled definition instead, remove the folder $HOME/.policy_sentry/
IAM actions under secretsmanager service that have the resource type *:
[
    "secretsmanager:GetRandomPassword",
    "secretsmanager:ListSecrets"
]
(venv) reetasingh-ltm8:policy_sentry reetasingh$ 
```

**Comment by reetasingh** [comment]
```
(venv) reetasingh-ltm8:policy_sentry reetasingh$ policy_sentry query action-table --service secretsmanager  --resource-type '*' --access-level read
Using the Local IAM definition: /Users/reetasingh/.policy_sentry/iam-definition.json. To leverage the bundled definition instead, remove the folder $HOME/.policy_sentry/
secretsmanager READ actions that have the resource type *:
[
    "secretsmanager:GetRandomPassword"
]

```

**Comment by reetasingh** [comment]
```
venv) reetasingh-ltm8:policy_sentry reetasingh$ policy_sentry query action-table --service ssm --resource-type parameter
Using the Local IAM definition: /Users/reetasingh/.policy_sentry/iam-definition.json. To leverage the bundled definition instead, remove the folder $HOME/.policy_sentry/
IAM actions under ssm service that have the resource type parameter:
[
    "ssm:AddTagsToResource",
    "ssm:DeleteParameter",
    "ssm:DeleteParameters",
    "ssm:GetParameter",
    "ssm:GetParameterHistory",
    "ssm:GetParameters",
    "ssm:GetParametersByPath",
    "ssm:LabelParameterVersion",
    "ssm:ListTagsForResource",
    "ssm:PutParameter",
    "ssm:RemoveTagsFromResource"
]

```

**Comment by kmcquade** [comment]
@reetasingh this looks awesome!

I'm good to merge this. But first - can you modify your PR so it updates the documentation for the `query action-table` section? https://policy-sentry.readthedocs.io/en/latest/querying/action-table/

**Comment by reetasingh** [comment]
> https://policy-sentry.readthedocs.io/en/latest/querying/action-table/

@kmcquade updated docs https://github.com/salesforce/policy_sentry/pull/261/commits/f81e27c106be5c0995a4b900d4727c933a0335e4

**Review by kmcquade — APPROVED** [review]

**Pull request merged**